staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go

```diff
@@ -65,11 +67,13 @@ type WebhookAuthorizer struct {
 	unauthorizedTTL     time.Duration
 	retryBackoff        wait.Backoff
 	decisionOnError     authorizer.Decision
+	metrics             AuthorizerMetrics
 }
 
 // NewFromInterface creates a WebhookAuthorizer using the given subjectAccessReview client
-func NewFromInterface(subjectAccessReview authorizationv1client.SubjectAccessReviewInterface, authorizedTTL, unauthorizedTTL time.Duration, retryBackoff wait.Backoff) (*WebhookAuthorizer, error) {
-	return newWithBackoff(subjectAccessReview, authorizedTTL, unauthorizedTTL, retryBackoff)
+func NewFromInterface(subjectAccessReview authorizationv1client.AuthorizationV1Interface, authorizedTTL, unauthorizedTTL time.Duration, retryBackoff wait.Backoff, metrics AuthorizerMetrics) (*WebhookAuthorizer, error) {
+	subjectAccessReviewClient := &subjectAccessReviewV1Client{subjectAccessReview.RESTClient()}
```

Comment:
nit: do you need a variable for this? can't this just be constructed on line below?

Reply:
would that be deterministic?

https://github.com/kubernetes/kubernetes/blob/293e07a83666dfc50abb6d2ce23e7dea8c4a46cf/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L678

on the other hand reading an HTTP status code from an error doesn't sound good :)